src/main/kotlin/io/easybreezy/hr/model/event/Conditions.kt

```diff
@@ -0,0 +1,64 @@
+package io.easybreezy.hr.model.event
+
+import io.easybreezy.infrastructure.exposed.dao.Embeddable
+import io.easybreezy.infrastructure.exposed.dao.EmbeddableClass
+import io.easybreezy.infrastructure.exposed.dao.EmbeddableTable
+
+class Conditions private constructor() : Embeddable() {
+    var isPrivate by ConditionsTable.isPrivate
+        private set
+    var isFreeEntry by ConditionsTable.isFreeEntry
+        private set
+    var isRepeatable by ConditionsTable.isRepeatable
+        private set
+
+    companion object : EmbeddableClass<Conditions>(Conditions::class) {
+        fun create(isPrivate: Boolean, isFreeEntry: Boolean, isRepeatable: Boolean): Conditions {
+            val conditions = Conditions()
+            conditions.isPrivate = isPrivate
+            conditions.isFreeEntry = isFreeEntry
+            conditions.isRepeatable = isRepeatable
+            return conditions
+        }
+
+        override fun createInstance(): Conditions {
+            return Conditions()
+        }
+    }
+
+    fun makePrivate() {
+        require(!isPrivate) { "Already private" }
+        isPrivate = true
+    }
+
+    fun makePublic() {
+        require(isPrivate) { "Already public" }
+        isPrivate = false
+    }
+
+    fun closeEntry() {
+        require(isFreeEntry) { "Already closed entry" }
+        isFreeEntry = false
+    }
+
+    fun openEntry() {
+        require(!isFreeEntry) { "Already opened entry" }
```

Comment:
Do you think it's a required check? What's the problem when I try to open the event twice?

Reply:
I do those checks, because there are events on conditions fields. And if do open twice than user can receive two similar letters

Reply:
Hmm, ok but for me, it's very strong conditions, in my opinion, I skip changes if it has been accepted